src/main/java/org/ays/common/util/validation/NoTrailingOrLeadingSpacesValidator.java

```diff
@@ -0,0 +1,26 @@
+package org.ays.common.util.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+/**
+ * A custom validator implementation for the {@link NoTrailingOrLeadingSpaces} annotation.
+ * Validates whether the provided text does not have
+ * trailing or leading spaces.
+ */
+class NoTrailingOrLeadingSpacesValidator implements ConstraintValidator<NoTrailingOrLeadingSpaces, String> {
+
+    @Override
+    public boolean isValid(String text, ConstraintValidatorContext constraintValidatorContext) {
+
+        if (text.startsWith(" ") || text.endsWith(" ")) {
+            constraintValidatorContext.disableDefaultConstraintViolation();
+            constraintValidatorContext.buildConstraintViolationWithTemplate("cannot start or end with space")
+                    .addConstraintViolation();
+            return false;
+        }
```

Comment:
Tek bir kontrol yaptığı için böyle yazmak yerine varsayılan hata mesajını güncellemek daha faydalı olabilir. Ek olarak text'in olmadığı durumlarda bu kod hata verebilir ve yanlış yönlendirebilir. Bu sebeple if kontrolü eklemekte fayda var ✅

Reply:
ekledim